### PR TITLE
Up vk version in shader compilation

### LINKED ISSUE #2468: Support Vulkan 1.3 target environment in shader compilation
# Template

If you dont understand something just leave it.
If you can provide more detailed information than the template allows for, please ignore the template and present all of your findings.

*   Version of vulkano: Vulkan 1.3
*   OS: TODO
*   GPU (the selected PhysicalDevice): TODO
*   GPU Driver: TODO
*   Upload of a reasonably minimal complete `main.rs` file that demonstrates the issue: TODO

# Issue

Current target environment in shader compilation limits Vulkan version up to 1.2

### PR DESCRIPTION
**Pull request by maratik123**
This fixes #2468

1. [-] Update documentation to reflect any user-facing changes - in this repository.

2. [+] Make sure that the changes are covered by unit-tests.

3. [+] Run `cargo clippy` on the changes.

4. [+] Run `cargo +nightly fmt` on the changes.

5. [+] Please put changelog entries **in the description of this Pull Request**
   if knowledge of this change could be valuable to users. No need to put the
   entries to the changelog directly, they will be transferred to the changelog
   file by maintainers right after the Pull Request merge.

   Please remove any items from the template below that are not applicable.

6. [+] Describe in common words what is the purpose of this change, related
   Github Issues, and highlight important implementation aspects.

Changelog:
```markdown
### Additions
- Support for Vulkan 1.3 target environment in shader compilation.
````

### CODE REVIEW

File: vulkano-util/src/window.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use winit::{
 ///
 /// ## Examples
 ///
-/// ```
+/// ```no_run
```
Comment (maratik123) [inline vulkano-util/src/window.rs:19]: Fixes cargo clippy error:
https://rust-lang.github.io/rust-clippy/master/index.html#/test_attr_in_doctest

File: examples/deferred/frame/system.rs [hunk 2]
```diff
@@ -408,11 +408,12 @@ impl<'a> Frame<'a> {
     pub fn next_pass<'f>(&'f mut self) -> Option<Pass<'f, 'a>> {
         // This function reads `num_pass` increments its value, and returns a struct corresponding
         // to that pass that the user will be able to manipulate in order to customize the pass.
-        match {
```
Comment (maratik123) [inline examples/deferred/frame/system.rs:411]: Fixes https://rust-lang.github.io/rust-clippy/master/index.html#/blocks_in_conditions

**Comment by Rua** [comment]
The clippy warnings are fixed by #2462, so you don't need to fix those yourself.

**Comment by maratik123** [comment]
> The clippy warnings are fixed by #2462, so you don't need to fix those yourself.

PR was merged with upstream

**Pull request merged**